### PR TITLE
fixed the grid layout schema and fixed layout bugs

### LINKED ISSUE #787: grid-container JSON schema hints missing key info
- SQL Operations Studio Version:

Steps to Reproduce:
- Open a package.json for an extension in Ops Studio
- Contribute a dashboard tab of type `grid-container`
- follow prompts to add an array then the 1st object

Expected:
- Get prompted for properties like name, row and column 
Actual:
- Get no useful prompts. The intellisense returned seems inaccurate for this container type

**Expected schema-type prompt (taken from https://github.com/Microsoft/sqlopsstudio/pull/740)**
```
"name": "widget 1",
						"widget": {
							"webview-widget": {
								"id": "my-widget"	
							}
						},
						"row":0,
						"col":0
```

**Actual intellisense given**
image

**Sample JSON used to repro**
```json
"dashboard.tabs": [
            {
                "container": {
                    "grid-container": [
                        {
                            
                        }
                    ]
                }
            }
]
```

### PR DESCRIPTION
**Pull request by llali**
Fixes #787

Also fixed some layout issues related to task widget inside the nav container. 

Note: I changed the schema of the grid layout to get colspan and rowspan as strings. so now it supports '*' 
This will cause warnings in the package.json of the extensions that set those values as number. But loading the grid container for t he extension still works.

### CODE REVIEW

File: src/sql/parts/dashboard/containers/dashboardGridContainer.component.ts [hunk 1]
```diff
@@ -105,15 +106,15 @@ export class DashboardGridContainer extends DashboardTab implements OnDestroy, O
 		let content = this.getContent(row, col);
 		let colspan: string = '1';
 		if (content && content.colspan) {
-			colspan = content.colspan;
+			colspan = this.covertToNumber(content.colspan, this.cols.length).toString();
```
Comment (kburtram) [inline src/sql/parts/dashboard/containers/dashboardGridContainer.component.ts]: typo: `convertToNumber` .  Also, presumably convertToNumber won't ever return `undefined` based on how it's being used, right?

File: src/sql/parts/dashboard/pages/dashboardPageContribution.ts [hunk 2]
```diff
@@ -97,33 +97,43 @@ export function generateDashboardGridLayoutSchema(type?: 'database' | 'server',
...
-			icon: {
-				type: 'string'
+			row: {
+				type: 'number',
+				description: localize('rowNumber', "The row of the component in the grid")
 			},
-			provider: {
+			rowspan: {
+				type: 'string',
+				description: localize('rowSpan', "The rowspan of the component in the grid. Default value is 1. Use '*' to set to number of rows in teh grid.")
```
Comment (kburtram) [inline src/sql/parts/dashboard/pages/dashboardPageContribution.ts]: typo `teh` should be `the`

Comment (kburtram) [inline src/sql/parts/dashboard/pages/dashboardPageContribution.ts]: for localization resource IDs, please prefix with the component name, like `dashboardpage.rowspan`.  These IDs need to be unique across the codebase and terms like `rowSpan` can easily collide and cause errors during XLF export process.

File: src/sql/parts/dashboard/pages/dashboardPageContribution.ts [hunk 3]
```diff
@@ -97,33 +97,43 @@ export function generateDashboardGridLayoutSchema(type?: 'database' | 'server',
...
+			},
+			col: {
+				type: 'number',
+				description: localize('colNumber', "The column of the component in the grid")
+			},
+			colspan: {
+				type: 'string',
+				description: localize('colspan', "The colspan of the component in the grid. Default value is 1. Use '*' to set to number of columns in teh grid.")
```
Comment (kburtram) [inline src/sql/parts/dashboard/pages/dashboardPageContribution.ts]: `teh` => `the`

File: src/sql/parts/dashboard/containers/dashboardGridContainer.component.ts (hunk 1 above)
Comment (llali) [reply src/sql/parts/dashboard/containers/dashboardGridContainer.component.ts]: it returns 1 by default

---
In reply to: [172673606](https://github.com/Microsoft/sqlopsstudio/pull/847#discussion_r172673606) [](ancestors = 172673606)

**Pull request merged**